app/src/main/java/com/nervousfish/nervousfish/activities/VisualVerificationActivity.java

```diff
@@ -98,16 +100,18 @@ private void nextActivity() {
      * @param v The view of the button being clicked.
      */
     public void buttonAction(final View v) {
-        final String button = v.getContentDescription().toString();
+        final int button = Integer.parseInt(v.getContentDescription().toString());
         LOGGER.info("button '%s' clicked", button);
 
-        if (this.securityCode.length() > VisualVerificationActivity.SECURITY_CODE_LENGTH) {
+        if (this.numTaps > VisualVerificationActivity.SECURITY_CODE_LENGTH) {
             LOGGER.warn("Security code already long enough");
-        } else if (this.securityCode.length() + 1 == VisualVerificationActivity.SECURITY_CODE_LENGTH) {
+        } else if (this.numTaps + 1 == VisualVerificationActivity.SECURITY_CODE_LENGTH) {
             this.securityCode += button;
             LOGGER.info("final code is: %s", this.securityCode);
             this.nextActivity();
         } else {
+            this.numTaps++;
+            this.securityCode *= NUM_BUTTONS;
```

Comment:
See the PR description:
> This Pull Request adds to the repository an encoding for the pattern tapped in VisualVerificationActivity.
Each button has a value between 1 and 12.
Initially code = 0
Every tap: code = code x 12 + value